core/src/main/java/org/geysermc/geyser/translator/protocol/bedrock/BedrockMoveEntityAbsoluteTranslator.java

```diff
@@ -64,7 +64,7 @@ public void translate(GeyserSession session, MoveEntityAbsolutePacket packet) {
         }
 
         float y = packet.getPosition().getY();
-        if (ridingEntity instanceof BoatEntity) {
+        if (ridingEntity instanceof BoatEntity && !ridingEntity.isOnGround() && !ridingEntity.getPassengers().isEmpty()) {
```

Comment:
`!ridingEntity.getPassengers().isEmpty()`
Is this condition necessary? `ridingEntity.getPassengers()` should always contain the current session's player entity and removing this condition did not seem to change anything.

Same goes for the change in `BedrockPlayerInputTranslator`

Reply:
fair point; yea. Tested with a debug print; that is always true - removed it in new commit